required/latex-lab/latex-lab-math.dtx

```diff
@@ -685,8 +712,148 @@
+%    \begin{macrocode}
+\tl_const:Nn \c_@@_mathml_write_init_tl 
+  {
+    <!DOCTYPE~html>
+    \iow_newline:
+    <html>
+    \iow_newline:
+  }
+\tl_new:N \l_@@_mathml_write_before_tl  
+\tl_const:Nn \c_@@_mathml_write_after_tl 
+  {
+    \iow_newline:
+    </div>
+    \iow_newline:
+  }  
+\tl_const:Nn \c_@@_mathml_write_final_tl 
+  {
+    </html>
+  }  
+%    \end{macrocode}
+% \end{macro}
 
+% \begin{socketdecl}{tagsupport/math/mathml/write/prepare}
+% To prepare the hash and the starting command we use a socket, so 
+% that both the dummy and luamml can make use of it.
+%    \begin{macrocode}
+\socket_new:nn {tagsupport/math/mathml/write/prepare}{0}
+%    \end{macrocode}
+% \end{socketdecl}
+%
+% \begin{plugdecl}{On}
+%    \begin{macrocode}
+\socket_new_plug:nnn{tagsupport/math/mathml/write/prepare}{On}
+  {
+    \str_set:NV\l_@@_tmpa_str\l_@@_content_AF_source_tl
+    \str_replace_all:Nnn\l_@@_tmpa_str{&}{&amp;}
+    \str_replace_all:Nnn\l_@@_tmpa_str{<}{&lt;}
+    \tl_set:Nn \l_@@_mathml_write_before_tl 
+      {
+        <div>
+        \iow_newline:
+        <h2>\c_backslash_str mml\c_space_tl \int_use:N \g_@@_math_total_int </h2>
+        \iow_newline:
+        <p>\l_@@_tmpa_str</p>
+        \iow_newline:
+        <p>\l_@@_content_hash_tl </p>
+        \iow_newline:
+      }   
+  }
+%    \end{macrocode}
+% \end{plugdecl}
 %
+% With luatex we automatically generate mathml with luamml if the package
+% can be loaded and unicode-math is detected. 
+% We start the process in the begindocument/end hook
+% so that the reading from a previous compilation can happen before!
+%    \begin{macrocode}
+\sys_if_engine_luatex:T
+ {
+   \file_if_exist:nT { luamml.sty }
+     {
+       \RequirePackage  { luamml }
+       \AddToHook{begindocument/end}
+         { 
+           \bool_if:NT \g__tag_math_luamml_bool 
+             { 
+               \@ifpackageloaded { unicode-math }
+                 { \@@_luamml_activate_write: }
+                 { \msg_warning:nn { tag }{ unicode-math-missing } }               
+             } 
+         }
+     }
+ }
+\msg_new:nnn { tag }{ unicode-math-missing } 
+  {
+    The~package~unicode-math~is~missing\\
+    luamml~will~not~create~an~MathML~file.\\
+    To~avoid~this~warning~load~unicode-math~or~disable~luamml\\
+    with~\tl_to_str:n{\tagpdfsetup{math/mathml/luamml=false}}
+  } 
+\cs_new_protected:Npn \@@_luamml_activate_write:
+ {
+%    \end{macrocode}
+% to avoid that nothing is written in the first run, we must activate the sockets:
+%    \begin{macrocode}
+   \bool_gset_true:N\g__tag_math_mathml_AF_bool
+   \AssignSocketPlug{tagsupport/math/struct/begin}{test-mathml}
+   \AssignSocketPlug{tagsupport/math/struct/end}{test-mathml}
+   \AssignSocketPlug{tagsupport/math/substruct/begin}{single}
+   \AssignSocketPlug{tagsupport/math/substruct/end}{single} 
+   \int_set:Nn \l__luamml_pretty_int { 5 }
+   \RegisterFamilyMapping\symsymbols{oms}
+   \RegisterFamilyMapping\symletters{oml}
```

Comment:
This is OK for initial merge but I think the fact that unicode-math leaves `\varCapitalGreek` defined and using legacy 8 bit fonts is a bug in that package we shoudl fix there. then these mappings would not be needed.

Reply:
One of many "strangenesses" -- but we already know that unicode-math needs a lot of work before we recommend its use for anything non-trivial.

Reply:
In practice unicode-math is pretty stable and we do need to recommend its use for all math tagging at least for the medium term, probably in the end we'll want some version loaded by default in lualatex formats but we are not there yet.

Reply:
@car222222 I'm not sure what you mean - as @davidcarlisle says, it's pretty stable and whilst some refinement may be desirable, some things may be constrained by decisions in the early 1990s (cf. ConTeXt, where they've gone entirely Unicode but with a very different approach to some aspects of stability). In my experience as a user, `unicode-math` 'just works'.

Reply:
@car222222  unicode-math is used wildly for lots of non-trivial mathematic and normally works fine. Only during the tests I found that it doesn't support all amsmath commands see https://github.com/latex3/unicode-math/issues/442

Reply:
Please can we see some of this wild, wide use of unicode-math? 

How does this relate to the fact that unicode-math itself documents many of its problems, etc? 
Are these concerns in fact not problems, or what? 

All very confusing!  But it works for everyone and everything, then no matter.

Reply:
@car222222  you could for example look at the WTPDF examples where all the math uses unicode-math including the arxiv papers, the Bohr collection, the amsmath guide etc. or for more "in the wild" use [stackexchange unicode-math tag](https://tex.stackexchange.com/questions/tagged/unicode-math)

Reply:
What does "look at" mean here?
I am sure they are exquisitely beautiful to look at, but so what?

Reply:
How can unicode-math be described as in any sense "stable" when it is full of warnings and notes on the many unfinished areas and unfixed problems.

Or am I looking at a different version thereof?